r/R/type.R

```diff
@@ -320,11 +320,27 @@ NestedType <- R6Class("NestedType", inherit = DataType)
 #' @return An Arrow type object inheriting from DataType.
 #' @export
 #' @seealso [dictionary()] for creating a dictionary (factor-like) type.
-#' @examplesIf arrow_available()
+#' @examplesIf arrow_available() && arrow_with_dataset()
```

Comment:
This will mean that we don't generate any of the examples if `arrow_with_dataset()` is `FALSE` though; even the ones which don't require Datasets.  How about using `\dontrun` instead for the `dplyr` block?

Reply:
I am wondering if it is sufficient to just remove this condition in the first place since #12533 was merged.

Reply:
I think we'd still have the issue of these examples failing on builds where dplyr isn't present, especially as it's not one of our imported packages.

Reply:
Sounds like a good idea to me then!